src/bundle.ts

```diff
@@ -70,6 +70,10 @@ const bundle = (document: unknown, bundleRoot: JsonPath, errorsRoot: JsonPath, k
     traverse(cur ? cur : objectToBundle, {
       onEnter: ({ value: parent }) => {
         if (hasRef(parent) && isLocalRef(parent.$ref)) {
+          if (parent.$ref === '#') {
+            throw new Error('it is never OK to reference the document itself from within the document');
+          }
```

Comment:
I added this as a fail-safe, so that we don't recurse infinitely.

Reply:
that is one tricky as well, because technically such a $ref is perfectly valid in JSON Schema, and bundle is used for JSON Schema models too.

Reply:
do we do any string manipulation like trimming whitespace after this? just want to make sure that if someone does `"  #           "` that won't break this logic

Reply:
Interestingly, when you add such a self-reference to a JSON Schema document, it's the file path that's referenced, not `#`.  That is, when I add a self-reference in Studio to a file named `widget.yaml`, this is the result:

```yaml
title: widget
x-stoplight:
  id: f0t0l2imlxn8y
type: object
properties:
  id:
    type: string
  subWidget:
    $ref: ./widget.yaml
```

This helps explain why this problem hasn't broken everything in the past (that we know of): you have to write it into the file "manually".